src/boot/efi/chid.c

```diff
@@ -0,0 +1,385 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024 Nikita Travkin <nikita@trvn.ru> */
+
+/*
+ * Based on Linaro dtbloader implementation.
+ * Copyright (c) 2019, Linaro. All rights reserved.
+ *
+ * https://github.com/aarch64-laptops/edk2/blob/dtbloader-app/EmbeddedPkg/Application/ConfigTableLoader/CHID.c
+ */
+
+#include "chid.h"
+#include "efi.h"
+#include "proto/Hash2.h"
+#include "util.h"
+#include "vmm.h"
+
+/**
+ * hash_strings() - Hash a list of strings after concatenating them.
+ * @alg:      Algorighm to use from HASH_PROTOCOL.
+ * @hash:     Pointer to the result buffer.
+ * @seed:     Arbitrary data to prepend to payload.
+ * @seed_len: Size of seed.
+ * @count:    Amount of strings.
+ * @...:  One or more char16_t strings to use as message.
+ */
+static EFI_STATUS hash_strings_sha1(EFI_SHA1_HASH2 *hash, uint8_t *seed, int seed_len, int count, ...) {
```

Comment:
why is the seed not const?

Reply:
why an explicit "count", rather than just terminating the list with a sentinel NULL pointer?

Reply:
the seed is just "abritrary bits", no? should be `const void*` hence, not typed.